### DISCUSSION TITLE
Digital product with above a 10k of variants

### DISCUSSION THREAD

**Discussion by deltacodepl** [saleor/saleor#11964] [Q&A] [answered]
Hi

I'm curious if there is a chance to use so great project like Saleor to create a website with digital products, which have a dozen of attributes and a couple of values for each?
Product variants will be count in thousands.
Therefore Fronted needs to have more traditional select options, but I presume that is feasible?

Thank you in advance

**Comment by karolkielecki** [comment, marked as answer]
Hi @deltacodepl! 

Saleor is headless -- you can connect any frontend you want. You can create as many attributes as you wish and assign many values to each. You can manage product data with a spreadsheet interface directly in the dashboard, bulk updates, and copy-paste from external sources - see [how it works here](https://www.linkedin.com/feed/update/urn:li:activity:6998654223252885504).

One of the Saleor projects with many product variants is [Rough Trade](https://roughtrade.com/).  
To manage a large number of products, variants, etc., you can also build an app (integration) with a PIM tool. Learn more in [our documentation](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts). 

The fastest and easiest way to test Saleor is to create a free developer account on our [Cloud](https://cloud.saleor.io/signup). 

Could you please tell us more about your use case? We'll be happy to answer your questions.
